server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridAggregationBuilder.java

```diff
@@ -101,20 +146,43 @@ protected AggregationBuilder shallowCopy(Builder factoriesBuilder, Map<String, O
      */
     public GeoGridAggregationBuilder(StreamInput in) throws IOException {
         super(in, ValuesSourceType.GEOPOINT, ValueType.GEOPOINT);
+        type = GeoHashType.readFromStream(in);
         precision = in.readVInt();
         requiredSize = in.readVInt();
         shardSize = in.readVInt();
     }
 
     @Override
     protected void innerWriteTo(StreamOutput out) throws IOException {
+        type.writeTo(out);
         out.writeVInt(precision);
         out.writeVInt(requiredSize);
         out.writeVInt(shardSize);
     }
 
+    public GeoGridAggregationBuilder type(String type) {
+        try {
+            this.type = GeoHashType.forString(type);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException(
+                "[type] is not valid. Allowed values: " +
+                    Stream.of(GeoHashType.values())
+                        .map(GeoHashType::toString)
+                        .collect(Collectors.joining(", ")) +
+                    ". Found [" + type + "] in [" + name + "]");
+        }
+        // Some tests bypass parsing steps, so keep the default precision consistent.
+        // Note that tests must set precision after setting the type
+        this.precision = this.type.getHandler().getDefaultPrecision();
+        return this;
+    }
```

Comment:
As I mentioned above I think we should set the type in the constructor so the precision can be set directly